test/functional/test_framework/test_node.py

```diff
@@ -174,11 +186,12 @@ def assert_start_raises_init_error(self, extra_args=None, expected_msg=None, par
 
         Will throw if bitcoind starts without an error.
         Will throw if an expected_msg is provided and it does not match bitcoind's stdout."""
-        with tempfile.SpooledTemporaryFile(max_size=2**16) as log_stderr:
+        with tempfile.NamedTemporaryFile(dir=os.path.join(self.datadir, 'stderr'), delete=False) as log_stderr, \
```

Comment:
Unable to replicate this bug on my machine, but the tests it fails on some machines is from these declarations. A small analysis - 

The error reported is `FileNotFoundError`, however, checking the code for creating temp files in python:
```
def _mkstemp_inner(dir, pre, suf, flags):
    """Code common to mkstemp, TemporaryFile, and NamedTemporaryFile."""

    names = _get_candidate_names()

    for seq in xrange(TMP_MAX):
        name = names.next()
        file = _os.path.join(dir, pre + name + suf)
        try:
            fd = _os.open(file, flags, 0600)
            _set_cloexec(fd)
            return (fd, _os.path.abspath(file))
        except OSError, e:
            if e.errno == _errno.EEXIST:
                continue # try again
            raise

    raise IOError, (_errno.EEXIST, "No usable temporary file name found")
```
Such an error could not have originated from this function. So the OS created the file but its not visible? Reading the docs gives this - 

```
Under Unix, the directory entry for the file is either not created at all or is removed immediately after the file is created. 
Other platforms do not support this; your code should not rely on a temporary file created using this function having or not having a visible name in the file system.
...
Whether the name can be used to open the file a second time, while the named temporary file is still open, varies across platforms (it can be so used on Unix; it cannot on Windows NT or later). 
```